### DISCUSSION TITLE
Accessing nested stacks which App.synth

### DISCUSSION THREAD

**Discussion by omriman12** [aws/aws-cdk#29215] [Q&A]
I'm trying to run Synth tests.
I'm doing this by running: `app_synth: CloudAssembly = app.synth()`
And verifying the resources are as expected..

Problem is, I have a nested stack, which I see within my root stack synth output, but I couldn't figure out how to access resources within the nested stack.

Using CDK version 2.128.0.
Python 3.8.15

**Comment by relationaldba** [comment]
I am facing the same issue. I use nested stacks and the Cloudformation template rendered in the CloudAssembly object appears to be missing resources from the nested stacks.

```python
app = App()
stack = MainStack(...)
nested_stack = NestedStack(stack,...)
cloud_assembly = app.synth()  
template = cloud_assembly.get_stack_artifact(stack.artifact_id).template
```

Please let me know if you found a solution, thanks
